frontend/public/components/utils/units.js

```diff
@@ -223,6 +294,14 @@ export const humanizeCpuCores = (v) => {
   };
 };
 export const humanizePercentage = (value) => {
+  if (value instanceof Array) {
+    value.forEach((point) => {
+      if (!isFinite(point.y)) {
+        point.y = 0;
+      }
+    });
+    return [value, '%'];
+  }
```

Comment:
These changes are causing a lot of typescript errors. I also don't think this is the best way to go. I don't think we should overload the humanize functions to handle an entire array. IMO, this is just confusing and will cause a TypeScript nightmare. Maybe we could provide a callback prop in the Area chart component that would allow the user to perform arbitrary mutations to the data before they are passed on to Victory. The return value of that callback could be an array containing the data set and the unit (as you have the humanize function doing here). The humanize functions already handle converting between units within a type by providing an `initialUnit` argument, so the unit returned by the data mutation callback could be passed on to the humanize function so that all values get displayed nicely on the y-axis and in tooltips.

Reply:
@spadgett What do you think about this? I'm not married to the idea of a new prop on the Area chart, because, as @bipuladh pointed out, this leads to broader changes and specialized uses for that component. I'm definitely open to other solutions, just wanted to point out that current implementation would not be optimal for type definitions/maintenance.

Reply:
I'm confused about why we need to overload all the humanize functions to take an array. Can't we just create a new function that expects an array and handle that separately?

Reply:
> Maybe we could provide a callback prop in the Area chart component that would allow the user to perform arbitrary mutations to the data before they are passed on to Victory.

I feel like we could replace the `humanize` prop with a generic metric type (memory or cpu) and let the area chart handle the logic.